File: C#/Graph/Graph/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Graph
{
    public class Node<TNode>
    {
        private TNode mValue;
        private string mName;
        private HashSet<Node<TNode>> mNeighbors;

        public Node(TNode value, string name)
        {
            mValue = value;
            mName = name;
            mNeighbors = new HashSet<Node<TNode>>();
        }

        public TNode GetValue() => mValue;

        public void SetValue(TNode value) => mValue = value;

        public string GetName() => mName;

        public void SetName(string name)
        {
            mName = name;
        }

        public void AddNeighbor(Node<TNode> node)
        {
            if (node != null) mNeighbors.Add(node);
        }

        public HashSet<Node<TNode>> GetNeighbors() => mNeighbors;
    }
}

File: src/graph/data/Node.java
package graph.data;

import java.util.HashSet;
import java.util.Set;

public class Node<NodeType> {
    private NodeType mValue;
    private String mName;
    private Set<Node<NodeType>> mNeighbors;

    public Node(NodeType value, String name) {
        mValue = value;
        mName = name;

        mNeighbors = new HashSet<Node<NodeType>>();
    }

    public NodeType GetValue() {
        return mValue;
    }

    public void SetValue(NodeType value) {
       this.mValue = value;
    }

    public String GetName() {
        return mName;
    }

    public void SetName(String name) {
        this.mName = name;
    }

    public void AddNeighbor(Node<NodeType> node) {
        mNeighbors.add(node);
    }

    public Set<Node<NodeType>> GetNeighbors() {
        return mNeighbors;
    }
}
